### DISCUSSION TITLE
Question on formatting metadata file for custom dataset

### DISCUSSION THREAD

**Discussion by Lonas13** [coqui-ai/TTS#2479] [General]
Hi, I'm very new to machine learning, I'm coming at this without any formal training while trying to educate myself so apologies in advance if this is an obvious question. 

I've successfully managed to install TTS and train the GlowTTS model on the ljspeech dataset; now I want to train my own dataset. I've formatted it identically to the ljspeech dataset vis a vis file structure, audio file size/order and metadate file structure.  However, I'm a bit confused because the documentation seems to imply all my metadata file needs is a .txt with two columns delineated with a pipe, one for the filename and another for transcription. However I ran into a range error when starting the trainer, and after some digging it turned out the ljspeech metadata file actually is a CSV file with three columns, the third being for normalized/phonomatic transcription. 

I don't see any steps in the documentation saying this is required, nor do I have a clue where I'd get that data from given I have almost ten thousand sample audio files. Any direction or advice would be appreciated. 

EDIT: SlashHarken's advice worked, but I also realized you can just add another pipe on the end of each row and copy over the text. Not sure if it matters for other models that don't use the GlowTTS config but even after editing the ljspeech function in formatters.py I ran into multiple issues with code breaking due to it missing that third data field.

**Comment by SlashHarken** [comment]


**Reply by Lonas13** [reply]
And that won't cause any  major issues down the road? I'm a little bit concerned about changing the code in the formatters.py file but I guess as long as I comment any changes I should always be able to go back and check on it in the future.

**Reply by fkwlqm** [reply]
Just copy that particular function in formatter.py, rename it `def cautious_af_mo_money_mo_problems():` and then use this new function in your `formatter_name` . Good luck. @Lonas13

**Reply by Lonas13** [reply]
Love it, thank you very much for the help!
